index/internal/tests/utils.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2025 EngFlow Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+type ExecConfig struct {
+	Dir string
+	Env []string
+}
+
+// Utility to execute commands
+func Execute(config ExecConfig, t *testing.T, program string, args ...string) exec.Cmd {
+	cmd := exec.Command(program, args...)
+	if config.Dir != "" {
+		cmd.Dir = config.Dir
+	}
+	if config.Env != nil {
+		cmd.Env = config.Env
+	}
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		t.Errorf("Failed to execute %v: %v", cmd.Args, err)
```

Comment:
I'd recommend calling `t.Fatalf` instead, but if the test should continue, return an error instead and let the caller decide whether it's a test failure.